docs/SIL.rst

```diff
@@ -1713,9 +1713,97 @@ normal uses always before consuming uses. Any such violations of ownership
 semantics would trigger a static SILVerifier error allowing us to know that we
 do not have any leaks or use-after-frees in the above code.
 
+Ownership Kind
+~~~~~~~~~~~~~~
+
 The semantics in the previous example is of just one form of ownership semantics
-supported: "owned" semantics. In SIL, we allow for values to have one of four
-different ownership kinds:
+supported: "owned" semantics. In SIL, we map these "ownership semantics" into a
+form that a compiler can reason about by mapping semantics onto a lattice with
+the following elements: `None`_, `Owned`_, `Guaranteed`_, `Unowned`_, `Any`_. We
+call this the lattice of "Ownership Kinds" and each individual value an
+"Ownership Kind". This lattice is defined as a 3-level lattice with::
+
+1. None being Top.
+2. Any being Bottom.
+3. All non-Any, non-None OwnershipKinds being defined as a mid-level elements of the lattice
+
+We can graphically represent the lattice via a diagram like the following::
+
+                +----+
+      +-------- |None| -----------+
+      |         +----+            |
+      |            |              |
+      v            v              v         ^
+  +-------+  +-----+------+  +---------+    |
+  | Owned |  | Guaranteed |  | Unowned |    +--- Value Ownership Kinds and
+  +-------+  +-----+------+  +---------+         Ownership Constraints
+      |            |              |
+      |            v              |         +--- Only Ownership Constraints
+      |          +---+            |         |
+      +--------->|Any|<-----------+         v
+                 +---+
+
+One moves down the lattice by performing a "meet" operation::
+
+  None meet OtherOwnershipKind -> OtherOwnershipKind
+  Unowned meet Owned -> Any
+  Owned meet Guaranteed -> Any
+
+and one moves up the lattice by performing a "join" operation, e.x.::
+
+  Any join OtherOwnershipKind -> OtherOwnershipKind
+  Owned join Any -> Owned
+  Owned join Guaranteed -> None
+
+This lattice is applied to SIL by requiring well formed SIL to:
+
+1. Define a static map of each SIL value to an OwnershipKind that classify the
```

Comment:
For some instructions, the ownership kind is static--every instance of that instruction has the same ownership, and for others it's dynamic--forwarding instructions are granted an ownership kind upon construction. Your point here, which is interesting, is that a given SILValue has constant ownership. Should say:
> Map each SIL value to its constant OwnershipKind...